src/ai/backend/manager/api/vfolder.py

```diff
@@ -113,35 +113,59 @@ async def ensure_vfolder_status(
 
-    if perm == VFolderAccessStatus.READABLE:
-        # if READABLE access status is requested, all operation status is accepted.
-        vf_status_conds = vfolders.c.status.in_(
-            [
-                VFolderOperationStatus.READY,
-                VFolderOperationStatus.PERFORMING,
-                VFolderOperationStatus.CLONING,
-                VFolderOperationStatus.DELETING,
-                VFolderOperationStatus.MOUNTED,
-            ]
-        )
-    elif perm == VFolderAccessStatus.UPDATABLE:
-        # if UPDATABLE access status is requested, only READY operation status is accepted.
-        vf_status_conds = vfolders.c.status == VFolderOperationStatus.READY
-    else:
-        # Otherwise, raise VFolderFilterStatusNotAvailable()
-        raise VFolderFilterStatusNotAvailable()
+    available_vf_statuses = set()
+    match perm:
+        case VFolderAccessStatus.READABLE:
+            # if READABLE access status is requested, all operation statuses are accepted.
+            available_vf_statuses = set(
+                [
+                    VFolderOperationStatus.READY,
+                    VFolderOperationStatus.PERFORMING,
+                    VFolderOperationStatus.CLONING,
+                    VFolderOperationStatus.DELETING,
+                    VFolderOperationStatus.MOUNTED,
+                ]
+            )
+        case VFolderAccessStatus.UPDATABLE:
+            # if UPDATABLE access status is requested, READY and MOUNTED operation statuses are accepted.
+            available_vf_statuses = set(
+                [
+                    VFolderOperationStatus.READY,
+                    VFolderOperationStatus.MOUNTED,
+                ]
+            )
+        case VFolderAccessStatus.DELETABLE:
+            # if DELETABLE access status is requested, only READY operation status is accepted.
+            available_vf_statuses = set(
+                [
+                    VFolderOperationStatus.READY,
+                    VFolderOperationStatus.DELETING,
+                ]
+            )
+        case _:
+            # Otherwise, raise VFolderFilterStatusNotAvailable()
+            raise VFolderFilterStatusNotAvailable()
     async with root_ctx.db.begin_readonly() as conn:
         entries = await query_accessible_vfolders(
             conn,
             user_uuid,
             user_role=user_role,
             domain_name=domain_name,
             allowed_vfolder_types=allowed_vfolder_types,
-            extra_vf_conds=(vf_name_conds & vf_status_conds),
+            extra_vf_conds=vf_name_conds,
             allow_privileged_access=True,
         )
         if len(entries) == 0:
-            raise VFolderFilterStatusFailed()
+            raise VFolderFilterStatusFailed(
+                f"Cannot find any folder with the given identity ({folder_id = }, {folder_name = })"
+            )
+        for entry in entries:
+            if entry["status"] in available_vf_statuses:
+                break
```

Comment:
Hmm, I'm a little bit unsure about this logic: Is it okay to just say okay if any of the folder has desired status? Not to all the vFolders?

Reply:
I made a change. thanks!